tests/unit-tests/test-class-usage-tracking.php

```diff
@@ -0,0 +1,318 @@
+<?php
+
+class Sensei_Usage_Tracking_Test extends WP_UnitTestCase {
+	public function setup() {
+		parent::setup();
+		$this->factory = new Sensei_Factory();
+		$this->usage_tracking = new Sensei_Usage_Tracking();
+	}
+
+	public function teardown() {
+		parent::teardown();
+	}
+
+	/**
+	 * Ensure cron job action is set up.
+	 */
+	public function test_cron_job_action_added() {
+		$this->usage_tracking->hook();
+		$this->assertTrue( !! has_action( 'sensei_core_jobs_usage_tracking_send_data', array( $this->usage_tracking, 'maybe_send_usage_data' ) ) );
+	}
+
+	/**
+	 * Ensure scheduling function works properly.
+	 */
+	public function test_maybe_schedule_tracking_task() {
+		// Make sure it's cleared initially
+		wp_clear_scheduled_hook( 'sensei_core_jobs_usage_tracking_send_data' );
+
+		// Record how many times the event is scheduled
+		$event_count = 0;
+		add_filter( 'schedule_event', function( $event ) use ( &$event_count ) {
+			if ( $event->hook === 'sensei_core_jobs_usage_tracking_send_data' ) {
+				$event_count++;
+			}
+			return $event;
+		} );
+
+		// Should successfully schedule the task
+		$this->assertFalse( wp_get_schedule( 'sensei_core_jobs_usage_tracking_send_data' ) );
```

Comment:
When doing multiple assertions in a test, it's helpful to add a comment as the last parameter that indicates what the assertion is testing (see https://github.com/Automattic/sensei/pull/1983/files#diff-b1c3ae0ceedad75266d017a21c49a56fR28). Otherwise, if an assertion in a test case fails, you'll need to spend time pinpointing which one it was.

Reply:
Good call. Messages added.